sql_generators/glean_usage/templates/events_stream_v1.query.sql

```diff
@@ -135,7 +135,25 @@ SELECT
 FROM
   base
 CROSS JOIN
+  -- See https://mozilla-hub.atlassian.net/browse/DENG-8432
+  -- Filtering out nimbus and normandy events that are emitted due to 'invalid-feature'.
+  -- The number of these events is too large to be processed, invalid-feature has also
+  -- been removed in more recent versions.
+  {% if app_name == "firefox_desktop_background_update" %}
+  ARRAY(
+    SELECT event
+    FROM UNNEST(events) AS event
+    WHERE NOT (
+      app_version_major = 138 AND app_version_minor IN (0, 1)
+      AND (
+        (event.category = 'nimbus_events' AND event.name = 'validation_failed')
+        OR (event.category = 'normandy' AND event.name = 'validation_failed_nimbus_experiment')
+      )
+    )
+  )
+  {% else %}
   UNNEST(events) AS event
+  {% endif %}
```

Comment:
In summary, I'd suggest something like this:
```suggestion
JOIN
  UNNEST(events) AS event
  {% if app_name == "firefox_desktop_background_update" %}
    -- See https://mozilla-hub.atlassian.net/browse/DENG-8432
    -- Filtering out nimbus and normandy events that are emitted due to 'invalid-feature'.
    -- The number of these events is too large to be processed, invalid-feature has also
    -- been removed in more recent versions.
    ON (
      (app_version_major = 138 AND app_version_minor IN (0, 1))
      AND (
        (event.category = 'nimbus_events' AND event.name = 'validation_failed')
        OR (event.category = 'normandy' AND event.name = 'validation_failed_nimbus_experiment')
      )
      AND STRUCT('reason' AS key, 'invalid-feature' AS value) IN UNNEST(event.extra)
    ) IS NOT TRUE
  {% endif %}
```

Reply:
I just tested this query, and it also exceeds resource limits after an hour of runtime.